### PR TITLE
update buildkite

### PR DESCRIPTION
**Pull request by simonbyrne**
## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by juliasloan25** [comment]
GHA are failing because we're still using NCDatasets' old syntax on this branch. #380 updates it - once that's merged you can rebase to fix the index errors

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,32 +1,30 @@
...
       - "julia --project -e 'using Pkg; Pkg.instantiate(;verbose=true)'"
-      - "julia --project -e 'using Pkg; Pkg.precompile()'"
       - "julia --project -e 'using Pkg; Pkg.status()'"
 
+      - echo "--- Instantiate test"
       - "julia --project=experiments -e 'using Pkg; Pkg.instantiate(;verbose=true)'"
-      - "julia --project=experiments -e 'using Pkg; Pkg.precompile()'"
       - "julia --project=experiments -e 'using Pkg; Pkg.status()'"
+    agents:
+      slurm_cpus_per_task: 8
```
Comment (simonbyrne) [inline .buildkite/pipeline.yml:25]:
```suggestion
      slurm_cpus_per_task: 8
      slurm_mem_per_cpu: 8G
```

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -58,3 +57,6 @@ steps:
       - label: "Water conservation"
         command: "julia --color=yes --project=experiments experiments/standalone/Soil/water_conservation.jl"
         artifact_paths: "experiments/standalone/Soil/water_conservation*png"
+  - wait
+
+  - command: "compact_depot"
```
Comment (simonbyrne) [inline .buildkite/pipeline.yml:62]:
```suggestion
  - command: "compact_depot"

```

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -1,32 +1,31 @@
...
-      - "julia --project -e 'using Pkg; Pkg.precompile()'"
       - "julia --project -e 'using Pkg; Pkg.status()'"
 
+      - echo "--- Instantiate test"
       - "julia --project=experiments -e 'using Pkg; Pkg.instantiate(;verbose=true)'"
-      - "julia --project=experiments -e 'using Pkg; Pkg.precompile()'"
       - "julia --project=experiments -e 'using Pkg; Pkg.status()'"
+    agents:
+      slurm_cpus_per_task: 8
+      slurm_mem_per_cpu: 8G
```
Comment (juliasloan25) [inline .buildkite/pipeline.yml:26]: why did you increase the memory here?

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (juliasloan25) [inline .buildkite/pipeline.yml:62]: does this cache the artifacts from this job on the cluster? do we need a DepotClear for ClimaLSM in buildkite now?

**Review by juliasloan25 — APPROVED** [review]

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (simonbyrne) [reply .buildkite/pipeline.yml:26]: It seems to speed up the compilation a bit.

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (simonbyrne) [reply .buildkite/pipeline.yml:62]: it's different than the shared depot we use in ClimaAtmos: basically in this job we have 2 depots: the first 1 is "clean", and the 2nd one just contains artifacts. We install everything into the 1st one, but at the end of each build, we move all the artifacts to the 2nd one

Comment (juliasloan25) [reply .buildkite/pipeline.yml:62]: so then do these 2 depots persist across jobs so they can be reused?

Comment (juliasloan25) [reply .buildkite/pipeline.yml:62]: also, will we ever need to clear these depots or do any other manual management of them?

Comment (simonbyrne) [reply .buildkite/pipeline.yml:62]: the 2nd one persists, the 1st one does not

Comment (simonbyrne) [reply .buildkite/pipeline.yml:62]: > also, will we ever need to clear these depots or do any other manual management of them?

I hope not, but I can't say for sure.

**Pull request merged**